### PR TITLE
fix: [CI-3443]: Fixing regex for map keys

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
##### Summary:

Bug fix for https://harness.atlassian.net/browse/CI-3443

image

image

image

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/10-common/utils/StringUtils.ts [hunk 1]
```diff
@@ -53,6 +53,8 @@ export const regexName = /^[A-Za-z0-9_-][A-Za-z0-9 _-]*$/
 
 export const regexIdentifier = /^[a-zA-Z_.][0-9a-zA-Z_$]*$/
 
+export const keyRegexIdentifier = /^[a-zA-Z_][0-9a-zA-Z_$]+(\.[0-9a-zA-Z_$]+)*$/
```
Comment (vkbansal) [inline src/modules/10-common/utils/StringUtils.ts:56]: Add test cases

**Review by vkbansal — CHANGES_REQUESTED** [review]

File: src/modules/10-common/utils/StringUtils.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/10-common/utils/StringUtils.ts:56]: Done.

**Review by mtran7 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/__tests__/StepsValidateUtils.test.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const strings = {
   validation: {
     validStepIdRegex: 'Identifier can only contain alphanumerics and _',
     validStepNameRegex: 'Step Name can only contain alphanumerics, spaces, _ and -',
-    validKeyRegex: 'Keys can only contain alphanumerics and _',
+    validKeyRegex: 'Keys can only contain alphanumerics and _ with an optional dot(.) in the middle',
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/__tests__/StepsValidateUtils.test.ts:26]: Great point! I'll create one as soon as this change gets merged.

**Review by vkbansal — APPROVED** [review]

**Pull request merged**